File: 2 sum/solution.py
from typing import List


def two_sum(nums: List[int], target: int) -> List[int]:
	"""Return indices of the two numbers that sum to target."""

	seen = {}

	for idx, value in enumerate(nums):
		complement = target - value

		if complement in seen:
			return [seen[complement], idx]

		seen[value] = idx

	return [-1, -1]

File: 2 sum/solution.js
/**
 * Returns indices of the two numbers that add up to the target.
 * @param {number[]} nums
 * @param {number} target
 * @returns {[number, number]}
 */
function twoSum(nums, target) {
	const seen = new Map();

	for (let i = 0; i < nums.length; i += 1) {
		const complement = target - nums[i];

		if (seen.has(complement)) {
			return [seen.get(complement), i];
		}

		seen.set(nums[i], i);
	}

	return [-1, -1];
}

module.exports = { twoSum };
